src/databases/indexedDb/loadRecords.ts

```diff
@@ -121,16 +121,67 @@ export async function loadRecords(countUpdater: ((totalCount: number) => void) |
 
 	if (countUpdater) void updateCount(collectionForCount, countUpdater);
 
-	if (needsMemorySort && sort) {
-		let items: object[];
-
-		if (sort.desc) processedCollection = processedCollection.reverse();
-		items = await processedCollection.sortBy(sort.id);
+	// Determine if there are out-of-line keys
+	const primKeySchema = table.schema.primKey;
+	const keyPath = primKeySchema.keyPath;
+	const isOutboundKey = (keyPath as string | string[] | null) === null;
 
-		const start = pageIndex * pageSize;
-		const end = Math.min(items.length, start + pageSize);
+	if (needsMemorySort && sort) {
+		// Check if we need to handle out-of-line keys
+		if (isOutboundKey) {
```

Comment:
We have renamed this `const`.

```suggestion
		if (hasOutOfLineKeys) {
```